src/request/swap-iframe/SwapIFrameApi.js

```diff
@@ -0,0 +1,385 @@
+                    value: storedRequest.redeem.input.witnessUtxo.value,
+                },
+                witnessScript: BitcoinJS.Buffer.from(parsedRequest.redeem.htlcScript),
+            }];
+
+            // The address is set in SignSwap after password entry.
+            const output = /** @type {{address: string, value: number}} */ (storedRequest.redeem.output);
+
+            // Construct transaction
+            const psbt = new BitcoinJS.Psbt({ network: BitcoinUtils.Network });
+
+            // Add inputs
+            // @ts-ignore Argument of type 'Uint8Array' is not assignable to parameter of type 'Buffer'.
+            psbt.addInputs(inputs);
+            // Add outputs
+            psbt.addOutput(output);
+
+            // Sign
+            const keyPair = BitcoinJS.ECPair.fromPrivateKey(
+                // @ts-ignore Argument of type 'import("...").Buffer' is not assignable to parameter of type 'Buffer'.
+                BitcoinJS.Buffer.from(privateKeys.btc[0], 'hex'),
+            );
+            psbt.signInput(0, keyPair);
+
+            // Verify that all inputs are signed
+            if (!psbt.validateSignaturesOfAllInputs()) {
+                throw new Error('Invalid or missing signature(s) for BTC transaction.');
+            }
+
+            // Finalize (with custom logic for the HTLC)
+            psbt.finalizeInput(0, (inputIndex, input /* , script, isSegwit, isP2SH, isP2WSH */) => {
+                if (!input.partialSig) {
+                    throw new Errors.KeyguardError('UNEXPECTED: Input does not have a partial signature');
+                }
+
+                if (!input.witnessScript) {
+                    throw new Errors.KeyguardError('UNEXPECTED: Input does not have a witnessScript');
+                }
+
+                const witnessBytes = BitcoinJS.script.fromASM([
+                    input.partialSig[0].signature.toString('hex'),
+                    input.partialSig[0].pubkey.toString('hex'),
+                    // Use zero-bytes as a dummy secret, required for signing
+                    '0000000000000000000000000000000000000000000000000000000000000000',
+                    'OP_1', // OP_1 (true) activates the redeem branch in the HTLC script
+                    input.witnessScript.toString('hex'),
+                ].join(' '));
+
+                const witnessStack = BitcoinJS.script.toStack(witnessBytes);
+
+                return {
+                    finalScriptSig: undefined,
+                    finalScriptWitness: HtlcUtils.witnessStackToScriptWitness(witnessStack),
+                };
+            });
+
+            // Extract tx
+            const tx = psbt.extractTransaction();
+
+            /** @type {KeyguardRequest.SignedBitcoinTransaction} */
+            result.btc = {
+                transactionHash: tx.getId(),
+                raw: tx.toHex(),
+            };
+        }
+
+        return result;
+    }
+}
+
+SwapIFrameApi.SESSION_STORAGE_KEY_PREFIX = 'swap_id_';
```

Comment:
This should be put into `Constants.js` as it is duplicated for `SignSwapApi.js` as the "same" constant.